builtin/providers/datadog/resource_datadog_user_test.go

```diff
@@ -0,0 +1,230 @@
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "name", "Test User"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDatadogUser_Email(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDatadogUserDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckDatadogUserConfigEmail,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDatadogUserExists("datadog_user.foo"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "email", "email@example.com"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "handle", "test@example.com"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "name", "Test User"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDatadogUser_IsAdmin(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDatadogUserDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckDatadogUserConfigIsAdmin,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDatadogUserExists("datadog_user.foo"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "email", "test@example.com"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "handle", "test@example.com"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "name", "Test User"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "is_admin", "true"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDatadogUser_NameUpdated(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckDatadogUserDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckDatadogUserConfigRequired,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDatadogUserExists("datadog_user.foo"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "email", "test@example.com"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "handle", "test@example.com"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "name", "Test User"),
+				),
+			},
+			resource.TestStep{
+				Config: testAccCheckDatadogUserConfigNameUpdated,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDatadogUserExists("datadog_user.foo"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "email", "test@example.com"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "handle", "test@example.com"),
+					resource.TestCheckResourceAttr(
+						"datadog_user.foo", "name", "Updated User"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDatadogUser_Role(t *testing.T) {
+	t.Skip("Datadog API does not support updating User role.")
```

Comment:
Not sure if this code is worth having if we can't use it. Was the the rationale to ad it?